worlds/generic/docs/RunFromSourceGuideForMac.md

```diff
@@ -0,0 +1,28 @@
+# Prerequisite Software
+Here is a list of software to install and source code to download.
+1. Python 3.8 or newer from the [macOS Python downloads page](https://www.python.org/downloads/macos/).
+2. Xcode from the [macOS App Store](https://apps.apple.com/us/app/xcode/id497799835).
+3. The source code from the [Archipelago Releases page](https://github.com/ArchipelagoMW/Archipelago/releases).
+4. The asset with darwin in the name from the [SNI Github](https://github.com/alttpo/sni/releases).
+5. If you would like to generate Enemized seeds for ALTTP locally (not on the website), you may need the EnemizerCLI from its [Github Page](https://github.com/Ijwu/Enemizer/releases).
+6. An Emulator of your choosing for games that need an emulator. For SNES games, I recommend RetroArch, entirely because it was the easiest for me to setup on macOS. It can be downloaded at the [RetroArch downloads page](https://www.retroarch.com/?page=platforms)
+## Setting up a virtual environment
+It is generally recommended that you use a virtual environment to run python based software to avoid contamination that can break some software. If Archipelago is the only piece of software you use that runs from python source code however, it is not necessary to use a virtual environment. 
+1. Run the command `python3 -m venv /path` to create a virtual environment. Running this command will create a new directory at the specified path, so make sure that path is clear for a new directory to be created.
```

Comment:
Historically people have been using a directory called "venv" inside the project.
So that means you create/navigate to the Archipelago directory and run `python3 -m venv venv` in there (if the path does not start with a `/` it is relative to the current directory), followed by `source venv/bin/activate`.

Reply:
I did not know that was the tradition. The document has been reworked accordingly. I may not be super familiar with UNIX CLI, but that is one of the things I do remember from my time deeper in CS and Software Engineering. :)